2024-05-31-notes.md

```diff
@@ -0,0 +1,189 @@
+# 2024-05-31 - Jess and Orion notes
+
+Things we need to handle in version 2.0:
+
+- base properties for nodes
+- embed attached files
+
+- "extensions" through schemas
+
+- the schema doesn't have a single high-level version number. rather, each of the nodes and relations is defined by a schema with a version
+  - every node has a schema associated with it
+  - the base spec "includes" all of the node types in `@opencanvas/*`
+  - all of the schemas are currently embedded in file itself
+  - `x` and `y` are required when implementing a node type, optionally `z`
+
+Node Base class
+
+```ts
+type Node = {
+  id: string;
+  schema: string;
+  schema_version: string;
+  x: number;
+  y: number;
+  z?: number;
+};
+```
+
+## What node extensions are supported in the initial spec?
+
+- [x] text
+- [x] shape
+  - (all are modeled with x, y, w, h, rotation)
+    - rotation in degrees, + or - between 360 and -360, positive is clockwise
+  - rectangle
+  - ellipse (circle)
+  - triangle
+  - diamond
+  - hexagon
+  - capsule (oval in tldraw)
+    - a rectangle where the shortest two edges of the object are semi-circles
+  - parallelogram
+  - star
+  - cloud
+  - heart
+  - arrow
+  - checkbox
+- [x] arrows (relations are separate)
+  - start: x, terminal
+  - bend: plus or minus pixels perpendicular from the midpoint of the arrow
+    - positive is to the left from the start to the end
+- [x] files
+  - `file-type`?
+    - support both embedded files and references to external sources (URLs, etc)
+  - Freehand
+
+## Relations
+
+- groups can be represented by a set relation
+  - relations can point to other relations in order to handle things like nested groups
+
+```json
+{
+  "nodes": {
+    "someShapeID11321": {
```

Comment:
`nodes` is an array in version 1.0, so why change it?

Reply:
No reason not to!